specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/SourceControls.json

```diff
@@ -230,6 +237,27 @@
             "required": true,
             "type": "string"
           },
+          {
+            "name": "x-ms-github-auxiliary",
+            "in": "header",
+            "description": "Github personal access token used for SourceControl.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "ignoreWorkflowDeletionFailure",
+            "in": "query",
+            "description": "Ignore Workflow Deletion Failure.",
+            "required": false,
+            "type": "boolean"
+          },
+          {
+            "name": "deleteWorkflow",
+            "in": "query",
+            "description": "Delete workflow.",
+            "required": false,
+            "type": "boolean"
+          },
```

Comment:
Please explain on what are these ?

If the API is deleting sourcecontrol resource, why are these needed ? 
Can it be a different API to delete workflows ?

Reply:
When creating source control resource, we will create a github workflow for customer. When deleting source control, we give customer the option to only delete source control resource or delete source control and created workflow together.

There is no need to delete workflows because it is a backend resource and not exposed to customer directly, which is also the design from day one.

Reply:
So, are these working APIs today that the customers are using  with ignoreWorkflowDeletionFailure and deleteWorkflow query params ? 

Delete is supposed to delete only a single resource.

Could you post any approval that these query parameters could be used ?

Reply:
In this case, we're deleting single resource (which is the SourceControl proxy resource). The option to delete the GH workflow file is just a consolidated option for convenience for the user to cleanup any leftover files.

Reply:
[ARMBlockingComment] If GH work flow is implicitly created on a source control creation, the deletion of source control resource should also result in deleting GH workflow . There should not be an extra query param for user to pass to clean up GH workflow. 

Please schedule a slot in ARM API Review office hours that happens every day to discuss further on this.
https://eng.ms/docs/products/arm/introduction/office_hours

Reply:
Did you get a review for this matter?

Reply:
yes